### DISCUSSION TITLE
[Question] How does the expand=True (or int) even work?

### DISCUSSION THREAD

**Discussion by angrymidget4728** [flet-dev/flet#2654] [Q&A]
### Question

I was initially under the impression that the `expand` property is a jacked-up/combined version of certain css properties, and while that is true to some extent, it doesn't seem flexible (assuming I'm using it correctly).

The following code gives what I wanted.
```python
    content1 = ft.Container(
        expand=1, 
        bgcolor=ft.colors.GREY_900, 
        margin=ft.margin.only(20,0,20,20), 
        border_radius=8,
        # content=ft.Text('item', expand=1,)
    )
    
    page.add(tabs)
    page.add(content1)
```

![image](https://github.com/flet-dev/flet/assets/139581466/0e014c1d-c9b0-40e6-ac4c-edb35b1596aa)

But as soon as add content, it shrinks the column to fit the content.

```python
    content1 = ft.Container(
        expand=1, 
        bgcolor=ft.colors.GREY_900, 
        margin=ft.margin.only(20,0,20,20), 
        border_radius=8,
        content=ft.Text('item', expand=1,)
    )
    
    page.add(tabs)
    page.add(content1)
```
![image](https://github.com/flet-dev/flet/assets/139581466/1ab8f97b-9fa1-4839-855d-7d3850e20052)

I recognize that this is the same behavior that occurs for `Row()` (since the tabs are in a row) but in such a case it is causing more problems than fix. My immediate thought was to use `page.width` but I realized that wouldn't be the same as the css `vh` measurement that constantly updates according to window size. Is using `page.width`/`page.on_window_event` with `page.update()` my only way or is there a more elegant/built-in way to tackle this using just the `expand` property?

EDIT:
As an analogy, practically any element in Figma has three options for both height and width each: fixed width, fill container, hug contents. From my test of Flet, the `expand=True` seems to apply 'fill container' for both height and width of a control when there is no content present. If you add content, then it takes its own liberty to make `height=fill container`, but `width=hug contents` (if in a column, and vice versa for row).
How do I take control of this behavior with Flet similar to Figma?

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by EmberLightVFX** [comment]
I don't know why but if I added `alignment=ft.alignment.top_left,` to content1 it works as I would expect

```
    content1 = ft.Container(
        expand=1, 
        alignment=ft.alignment.top_left,
        bgcolor=ft.colors.GREY_900, 
        margin=ft.margin.only(20,0,20,20), 
        border_radius=8,
        content=ft.Text('item', expand=1,)
    )
    
    page.add(content1)
```
